lib/rabbit_feed/event_definitions.rb

```diff
@@ -51,19 +51,31 @@ def defined_as &block
         @definition = block.call if block.present?
       end
 
-      def payload
-        ([
+      def payload_schema
+        { name: "#{name}_payload", type: 'record', fields: fields.map(&:schema) }
+      end
+
+      def metadata_schema
+        { name: 'event_metadata', type: 'record', fields: [
           (Field.new 'application',    'string', 'The name of the application that created the event'),
           (Field.new 'host',           'string', 'The hostname of the server on which the event was created'),
           (Field.new 'environment',    'string', 'The environment in which the event was created'),
-          (Field.new 'version',        'string', 'The version of the event'),
+          (Field.new 'version',        'string', 'The version of the event payload'),
+          (Field.new 'gem_version',    'string', 'The version of rabbit_feed by which the event was created'),
```

Comment:
> I think we might need a version for the event itself?. as for when it is needed to call from_pre_2_0?. I guess you are using the gem_version

Yes, that is the reason I've introduced the gem_version

> I see it as a different thing from the event structure itself?

Could you please give an example? I don't follow